webhook/core/pod_webhook_test.go

```diff
@@ -153,6 +192,54 @@ func TestPodResourceInjector_Handle(t *testing.T) {
 	}
 }
 
+// TestFargatePodResourceInjector_Handle test webhook annotating fargate pod with sg list.
+func TestFargatePodResourceInjector_Handle(t *testing.T) {
+	pod := NewFargatePod("test", "test_sa", "test_namespace")
+	assert.True(t, pod.ObjectMeta.Labels[fargatePodLabel] == "test_fp")
+	injector := getFargatePodResourceInjectorWithSGP()
+	resp := getResponse(pod, injector)
+	assert.True(t, resp.Allowed)
+	for _, p := range resp.Patches {
+		assert.True(t, p.Operation == "add")
+
+		pv := p.Value.(map[string]interface{})
+		assert.True(t, pv[fargatePodSgAnnotKey] == "sg-00001")
+	}
+}
+
+// TestFargatePodResourceInjectorWithoutSGP_WithAnnotPodSpec_Handle test webhook overriding fargate pod annotation with sg list.
+func TestFargatePodResourceInjectorWithoutSGP_WithAnnotPodSpec_Handle(t *testing.T) {
+	pod := NewFargatePod("test", "test_sa", "test_namespace")
+	annotMap := pod.ObjectMeta.Annotations
+	if annotMap == nil {
+		annotMap = make(map[string]string)
+	}
+	annotMap[fargatePodSgAnnotKey] = "sg-123"
+	pod.ObjectMeta.Annotations = annotMap
+	assert.True(t, pod.ObjectMeta.Labels[fargatePodLabel] == "test_fp")
+	injector := getFargatePodResourceInjectorWithoutSGP()
+	resp := getResponse(pod, injector)
+	assert.True(t, resp.Allowed)
+	for _, p := range resp.Patches {
+		assert.True(t, p.Operation == "remove")
```

Comment:
I think the test case is missing check that the annotation is successfully removed in the response?